src/proxy/ProtocolProbeSessionAccept.cc

```diff
@@ -136,7 +136,12 @@ struct ProtocolProbeTrampoline : public Continuation, public ProtocolProbeSessio
     } // end of Proxy Protocol processing
 
     if (proto_is_http2(reader)) {
-      key = PROTO_HTTP2;
+      if (netvc->get_service<TLSBasicSupport>() == nullptr) {
+        key = PROTO_HTTP2;
+      } else {
+        Dbg(dbg_ctl_http, "HTTP/2 prior knowledge was used on a TLS connection (protocol violation). Selecting HTTP/1 instead.");
+        key = PROTO_HTTP;
+      }
```

Comment:
While we're at it, I suggest adding a comment referencing the RFC. Maybe something like:

```cpp
   // RFC 9113 Section 3.3: Prior knowledge is only permissible for HTTP/2 over plaintext (non-TLS) connections.
```

Reply:
Added the comment as suggested.